sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -158,6 +158,17 @@ trait HashJoin extends JoinCodegenSupport {
         output, (streamedPlan.output ++ buildPlan.output).map(_.withNullability(true)))
   }
 
+  // Exposed for testing
+  @transient lazy val ignoreDuplicatedKey = joinType match {
+    case LeftExistence(_) =>
+      // For building hash relation, ignore duplicated rows with same join keys if:
+      // 1. Join condition is empty.
+      // 2. Join condition only references streamed attributes and build join keys.
```

Comment:
We only need to meet any one requirement? So maybe say 

```
1. Join condition is empty, or
2. ...
```

Reply:
@viirya - yes, updated.